Integrity hash file (#1024)
ariga/atlas: doc/md/concepts/integrity_hash_file.md

---

> id: hash_file

Reviewer: It's a sum_file that contains hashes. 
```suggestion
id: sum_file
```

slug: /concepts/integrity_hash_file

> title: Integrity Hash File

Reviewer: ```suggestion
title: Integrity Sum File
```

---

> ## Version Migration Files
> Version migration files are files that describe changes to the databases. Usually these files are prefixed with a number - which describe the version of the database change. The framework that runs these files uses these numbers to determine which changes have been applied to the database.

Reviewer: ```suggestion
## Version Migration Files
Version migration files are files that describe changes to the databases. Usually, these files are prefixed with a number that describes the version of the database change. The framework that executes these script files uses these numbers to determine which files have been applied to the database.
```

> In Ent, you’ll find these files in ent/migrations/ folder.

Author: ```suggestion

```

> In Django, you’ll find these files in project_name/app_name/migrations/ folder.

Reviewer: Why Ent and Django are mentioned here? Feels redundant to me.

Author: Will remove this in a later commit.

> In Django, you’ll find these files in project_name/app_name/migrations/ folder.

Author: ```suggestion

```

> In Django, you’ll find these files in project_name/app_name/migrations/ folder.

Author: ```suggestion

```

## Integrity Hash File

> Databases hold one of the most valuable assets of the organization. Unfortunately, databases are also quite fragile. Especially when working in teams where developers are working in parallel on the same business logic. Data loss is real and that’s why good practice is so important when working with DBs.

Reviewer: Maybe it's just my preferences, but I'd go with the "cut to the chase" approach:

> When multiple engineers work on the same project in parallel, there is a chance that their migration scripts will conflict with each other and may cause unexpected behavior. For example: ...
> ...
>
> Atlas provides a built-in approach to tackle this issue: a migration sum file ...
>
> ### Atlas Sum file
> \<more details about sum files\>
> ..

> Databases hold one of the most valuable assets of the organization. Unfortunately, databases are also quite fragile. Especially when working in teams where developers are working in parallel on the same business logic. Data loss is real and that’s why good practice is so important when working with DBs.

Author: ```suggestion
When multiple engineers work on the same project in parallel, there is a chance that their migration scripts will conflict with each other and may cause unexpected behavior. For example:
```

> There are 3 issues that can occur with version migration files.

Author: ```suggestion

```
